website/blog/2023-04-08-databend-weekly-88.md

```diff
@@ -0,0 +1,184 @@
+                    |
+                    eager group-by: eager SUM(x), eager count: cnt
+(2) Eager Split:
+                expression
+                    |
+     final aggregate: SUM(eager SUM(x) * cnt2), SUM(eager SUM(y) * cnt1)
+                    |
+                    join
+                    | \
+                    | eager group-by: eager SUM(y), eager count: cnt2
+                    |
+                    eager group-by: eager SUM(x), eager count: cnt1
+```
+
+If you are interested in learning more, please check out the resources listed below.
+
+- [PR #10716 | feat(planner): support eager aggregation](https://github.com/datafuselabs/databend/pull/10716)
+- [Paper | Eager Aggregation and Lazy Aggregation](https://www.vldb.org/conf/1995/P345.PDF)
+
+### Support All TPC-DS Queries
+
+Databend now supports all TPC-DS queries!
+
+TPC-DS is a decision support benchmark that models several generally applicable aspects of a decision support system, including queries and data maintenance. The benchmark provides a representative evaluation of performance as a general-purpose decision support system.
+
+If you are interested in learning more, please check out the resources listed below.
+
+- [PR | feat: support cte in dphyp](https://github.com/datafuselabs/databend/pull/10673)
+- [PR | chore(test): add tpcds to test](https://github.com/datafuselabs/databend/pull/10673)
+
+## Code Corner
+
+Discover some fascinating code snippets or projects that showcase our work or learning journey.
+
+### databend-driver - A driver for Databend in Rust
+
+The Databend community has developed a driver for Databend in Rust. It allows developers to easily connect to Databend and execute SQL queries using Rust.
+
+Here's an example of how to use the driver:
+
+```Rust
+use databend_driver::new_connection;
+
+let dsn = "databend://root:@localhost:8000/default?sslmode=disable";
+let conn = new_connection(dsn).unwrap();
+
+let sql_create = "CREATE TABLE books (
+    title VARCHAR,
+    author VARCHAR,
+    date Date
+);";
+conn.exec(sql_create).await.unwrap();
+let sql_insert = "INSERT INTO books VALUES ('The Little Prince', 'Antoine de Saint-Exupéry', '1943-04-06');";
+conn.exec(sql_insert).await.unwrap();
+```
+
+Welcome to try it out and give us feedback. If you want to learn more information, you can also follow the resources listed below.
+
+- [Crates.io - databend-driver](https://crates.io/crates/databend-driver)
+- [Github - datafuselabs/databend-client](https://github.com/datafuselabs/databend-client)
+
+### AskBend - SQL-based Knowledge Base Search and Completion
+
+AskBend is a Rust project that utilizes the power of Databend and OpenAI to create a SQL-based knowledge base from Markdown files.
+
+With AskBend, you can easily search and retrieve the most relevant information to your queries using SQL. The project automatically generates document embeddings from the content, enabling users to quickly find the information they need.
```

Comment:
With AskBend, you can easily search and retrieve the most relevant information to your queries using SQL. The project automatically generates document embeddings from the content, enabling you to quickly find the information you need.